astropy/units/quantity.py

```diff
@@ -1769,26 +1824,30 @@ def _unquantify_allclose_arguments(actual, desired, rtol, atol):
     try:
         desired = desired.to(actual.unit)
     except UnitsError:
-        raise UnitsError("Units for 'desired' ({}) and 'actual' ({}) "
-                         "are not convertible"
-                         .format(desired.unit, actual.unit))
+        raise UnitsError(
+            f"Units for 'desired' ({desired.unit}) and 'actual' "
+            f"({actual.unit}) are not convertible"
+        )
 
     if atol is None:
-        # by default, we assume an absolute tolerance of 0
+        # By default, we assume an absolute tolerance of 0.  The default
```

Comment:
THis is not strictly true: values of 0 are treated as being allowed to have any unit.

Reply:
The code after that converts `atol` into a `Quantity`, and then tries to convert `atol` to the same units as `actual`.  That means that if I do `u.isclose(1 * u.m, 1.1 * u.m, atol=0)` then I get a `u.UnitsError`.  I think this way is a bit more robust, since it's more explicit that the absolute tolerance does need consistent units.

Reply:
Ah, yes, that right, because we make a `Quantity` first, the 0 doesn't work. In any case, I agree that `None` is a clearer default here.